src/modules/admin/components/formRules/FormRulesCard.tsx

```diff
@@ -0,0 +1,101 @@
+import AddIcon from '@mui/icons-material/Add';
+import { Box, Button, Divider, Stack, Typography } from '@mui/material';
+
+import FormRuleTable from '../formRules/FormRuleTable';
+import TitleCard from '@/components/elements/TitleCard';
+import FormProjectMatchTable from '@/modules/admin/components/formRules/FormProjectMatchTable';
+import { useStaticFormDialog } from '@/modules/form/hooks/useStaticFormDialog';
+import {
+  CreateFormRuleDocument,
+  CreateFormRuleMutation,
+  FormRole,
+  FormRuleInput,
+  MutationCreateFormRuleArgs,
+  StaticFormRole,
+  useGetFormProjectMatchesQuery,
+} from '@/types/gqlTypes';
+import { evictQuery } from '@/utils/cacheUtil';
+
+interface Props {
+  formId: string;
+  formTitle: string;
+  formRole: FormRole;
+}
+const FormRulesCard: React.FC<Props> = ({ formTitle, formId, formRole }) => {
+  // Form dialog for adding new rules
+  const { openFormDialog, renderFormDialog } = useStaticFormDialog<
+    CreateFormRuleMutation,
+    MutationCreateFormRuleArgs
+  >({
+    formRole: StaticFormRole.FormRule,
+    mutationDocument: CreateFormRuleDocument,
+    localConstants: { formRole },
+    getErrors: (data) => data.createFormRule?.errors || [],
+    getVariables: (values) => ({
+      input: { input: values as FormRuleInput, definitionId: formId },
+    }),
+    onCompleted: () => {
+      evictQuery('formDefinition', { id: formId });
+      evictQuery('formRules');
+    },
+  });
+
+  // Fetch here in order to display the total number of project matches outside of the table.
+  // FormProjectMatchTable also requests this, so it will fetch from the apollo cache.
+  const { data: fetchFormData } = useGetFormProjectMatchesQuery({
+    variables: { id: formId },
+  });
```

Comment:
I am seeing this make 2 queries on page load
<img width="765" alt="Screenshot 2024-07-01 at 11 40 26 AM" src="https://github.com/greenriver/hmis-frontend/assets/5333982/434ca266-2228-4bde-95fe-fbf513d09977">

But I see why it's needed, since we don't have an easy way to get that number "out" of the GenericTableWithData component. There are definitely improvements that could be made to that component api, since we have this need elsewhere too.

This is OK to leave as-is but maybe we can set `limit: 0` explicitly since we're not doing anything with the returned nodes.

Reply:
Ah thanks for checking this. I updated the query to use `limit: 0` and updated the comment as well to be more honest.